src/main/java/jalasoft/com/models/Dictionary.java

```diff
@@ -0,0 +1,26 @@
+package jalasoft.com.models;
+
+import java.util.Set;
+
+public abstract class Dictionary<T> implements ISearch<T> {
+

```

Comment:
why a Dictionary has only a set of any type of items? shouldn't a Dictionary have a key and a value or something like that?